components/scream/src/share/atm_process/atmosphere_process.cpp

```diff
@@ -158,90 +158,64 @@ void AtmosphereProcess::set_computed_group (const FieldGroup& group) {
+        "  - Property check name: " + pc->name() + "\n"
+        "  - Atmosphere process MPI Rank: " + std::to_string(m_comm.rank()) + "\n");
+
+    // A Warn means we can continue, even if we can't repair,
+    // but we need to warn the user
+    if (check==Warn) {
+      std::cout << "WARNING: Pre run property check failed.\n"
+        "  - Property check name: " + pc->name() + "\n"
+        "  - Atmosphere process name: " + name() + "\n"
+        "  - Atmosphere process MPI Rank: " + std::to_string(m_comm.rank()) + "\n";
+    }
+
+    // If check did't pass, and we can repair, then do it.
+    if (check!=Pass && pc->can_repair()) {
+      pc->repair();
     }
   }
 }
 
-void AtmosphereProcess::check_computed_fields () {
-  // Loop over all the field checks on input fields, and execute them
-  for (const auto& fp_it : m_property_checks_out) {
-    const auto& fid   = fp_it.first;
-    const auto& fname = fid.name();
-    const auto& gname = fid.get_grid_name();
-    const auto& prop_check = *fp_it.second;
-    if (has_computed_field(fid)) {
-      auto& f = get_field_out(fname,gname);
-      const auto check = prop_check.check(f);
-      EKAT_REQUIRE_MSG(check || prop_check.can_repair(),
-          "Error! Output field property check failed (and cannot be repaired).\n"
-          "       Atm proc: " + this->name() + "\n"
-          "       Field:    " + fname + "\n"
-          "       Check:    " + prop_check.name() + "\n");
-      if (not check) {
-        prop_check.repair(f);
-      }
-    } else {
-      auto& group = get_group_out(fname,gname);
-      if (group.m_bundle) {
-        auto& f = *group.m_bundle;
-        const auto check = prop_check.check(f);
-        EKAT_REQUIRE_MSG(check || prop_check.can_repair(),
-            "Error! Output field property check failed (and cannot be repaired).\n"
-            "       Atm proc: " + this->name() + "\n"
-            "       Field:    " + group.m_info->m_group_name + "\n"
-            "       Check:    " + prop_check.name() + "\n");
-        if (not check) {
-          prop_check.repair(f);
-        }
-      } else {
-        for (const auto& it : group.m_fields) {
-          auto& f = *it.second;
-          const auto check = prop_check.check(f);
-          EKAT_REQUIRE_MSG(check || prop_check.can_repair(),
-              "Error! Output field property check failed (and cannot be repaired).\n"
-              "       Atm proc: " + this->name() + "\n"
-              "       Field:    " + it.first + "\n"
-              "       Check:    " + prop_check.name() + "\n");
-          if (not check) {
-            prop_check.repair(f);
-          }
-        }
-      }
+void AtmosphereProcess::run_property_checks_post () const {
+  constexpr auto Pass = PropertyCheck::CheckResult::Pass;
+  constexpr auto Warn = PropertyCheck::CheckResult::Warn;
+  constexpr auto Fail = PropertyCheck::CheckResult::Fail;
+
+  // Run all pre-condition property checks
+  for (const auto& pc : m_property_checks_pre) {
```

Comment:
This should probably be `m_property_checks_post`, right?

Reply:
Argh, good catch!